lib/src/collect.dart

```diff
@@ -227,26 +270,47 @@ Future<List<Map<String, dynamic>>> _getCoverageJson(VmService service,
         print('tokenPos $tokenPos has no line mapping for script $scriptUri');
         continue;
       }
-      hitMap[line] = hitMap.containsKey(line) ? hitMap[line]! + 1 : 1;
+      _increment(hits.lineHits, line);
+      if (hits.funcNames.containsKey(line)) _increment(hits.funcHits, line);
     }
     for (final tokenPos in coverage.misses!) {
       final line = _getLineFromTokenPos(script, tokenPos);
       if (line == null) {
         print('tokenPos $tokenPos has no line mapping for script $scriptUri');
         continue;
       }
-      hitMap.putIfAbsent(line, () => 0);
+      hits.lineHits.putIfAbsent(line, () => 0);
     }
+    hits.funcNames.forEach((line, funcName) {
+      hits.funcHits.putIfAbsent(line, () => 0);
+    });
   }
 
   // Output JSON
   final coverage = <Map<String, dynamic>>[];
-  hitMaps.forEach((uri, hitMap) {
-    coverage.add(toScriptCoverageJson(uri, hitMap));
+  hitMaps.forEach((uri, hits) {
+    coverage.add(toScriptCoverageJson(uri, hits));
   });
   return coverage;
 }
 
+void _increment(Map<int, int> counter, int key) {
```

Comment:
Can you make this more descriptive? I had to look up what it was doing.

Reply:
Done